src/umessage.rs

```diff
@@ -63,7 +62,22 @@ impl From<&str> for UMessageError {
     }
 }
 
+/// A container for a message's attributes and payload.
+#[derive(Debug, Clone)]
+pub struct UMessage {
+    attributes: UAttributes,
+    payload: Option<Bytes>,
+}
+
 impl UMessage {
+    pub fn attributes(&self) -> &UAttributes {
```

Comment:
I really like this!

So the idea is that we have now not generated the code for the UMessage .proto object, so we can _force_ correctness, as the only way to build a UMessage is via UMessageBuidler and we've made the struct fields private and can only grab the UAttributes and payload Bytes as reference.

Slick!!

Reply:
That _would have been_ the idea, yes. Sadly, though, UMessage is actually being used by uTwin in its input/output messages, so the initial approach will not work ...